TST: Replaced try-catch blocks with pytest.raises

ShaharNaveh opened the pull request:
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Commit:
TST: Replaced try-catch blocks with pytest.raises

Commit:
Merge remote-tracking branch 'upstream/master' into TST-try-catch

WillAyd commented on pandas/tests/util/test_assert_series_equal.py:
```diff
@@ -36,13 +36,9 @@ def _assert_not_series_equal(a, b, **kwargs):
         tm.assert_series_equal(a, b, **kwargs)
-        msg = "The two Series were equal when they shouldn't have been"
-
-        pytest.fail(msg=msg)
```
I don't think removing this is entirely equivalent; might be root of CI failures

jorisvandenbossche commented on pandas/tests/io/test_sql.py:
```diff
@@ -226,10 +226,8 @@ def _get_all_tables(self):
+        with pytest.raises(Error):
             self.conn.close()
-        except Error:
-            pass
```
I think this is kind of a "teardown" method of the tests, so not sure this should use an assert instead of try/except

jorisvandenbossche commented on pandas/tests/io/test_sql.py:
```diff
@@ -538,13 +536,12 @@ class DummyException(Exception):
         ins_sql = "INSERT INTO test_trans (A,B) VALUES (1, 'blah')"
-        try:
+
+        with pytest.raises(DummyException):
```
I don't think we need to assert the exception here, from reading the comment we are just raising it to ensure the transaction is correctly rolled back. I think using a try/except with ignoring the exception makes that clearer.

Commit:
Merge remote-tracking branch 'upstream/master' into TST-try-catch

Commit:
Reverted root cause of CI failure

Commit:
Applied @jorisvandenbossche suggestions

REFS:
https://github.com/pandas-dev/pandas/pull/31165#discussion_r369432765
https://github.com/pandas-dev/pandas/pull/31165#discussion_r369433351

Commit:
Fixed other CI errors

Commit:
Merge remote-tracking branch 'upstream/master' into TST-try-catch

jorisvandenbossche commented:
Thanks @MomIsBestFriend !
